packages/form-state-manager/src/manager-api/manager-api.ts

```diff
@@ -516,45 +518,92 @@ const createManagerApi: CreateManagerApi = ({
     runFormValidation = false;
   }
 
-  function handleFieldError(name: string, isValid: boolean, error: string | undefined = undefined, validating = false) {
+  function handleFieldError(name: string, isValid: boolean, error: string | undefined = undefined, validating = false, cacheKey: string) {
     const prevMeta = getFieldState(name)?.meta || ({} as Meta);
     const { error: prevError, valid: prevIsValid, validating: prevValidating } = prevMeta;
     if (error !== prevError || isValid !== prevIsValid || validating !== prevValidating) {
-      setFieldState(name, (prev: FieldState) => ({
-        ...prev,
-        meta: {
-          ...prev.meta,
-          error,
-          valid: isValid,
-          invalid: !isValid,
-          validating,
-          warning: undefined
+      setFieldState(name, (prev: FieldState) => {
+        const newState = {
+          ...prev,
+          meta: {
+            ...prev.meta,
+            error,
+            valid: isValid,
+            invalid: !isValid,
+            validating,
+            warning: undefined
+          }
         }
-      }));
+        validationCache.set(cacheKey, newState)
+        return newState
+      });
     }
 
     updateError(name, isValid ? undefined : error);
   }
 
-  function handleFieldWarning(name: string, warning: string | undefined = undefined, validating = false) {
+  function handleFieldWarning(name: string, warning: string | undefined = undefined, validating = false, cacheKey: string, ) {
     const prevMeta = getFieldState(name)?.meta || ({} as Meta);
     const { warning: prevWarning, validating: prevValidating } = prevMeta;
     if (warning !== prevWarning || validating !== prevValidating) {
-      setFieldState(name, (prev: FieldState) => ({
-        ...prev,
-        meta: {
-          ...prev.meta,
-          warning,
-          error: undefined,
-          valid: true,
-          invalid: false,
-          validating
-        }
-      }));
+      setFieldState(name, (prev: FieldState) => {        
+        const newState = {
+          ...prev,
+          meta: {
+            ...prev.meta,
+            warning,
+            error: undefined,
+            valid: true,
+            invalid: false,
+            validating
+          }}
+          validationCache.set(cacheKey, newState)
+        return newState});
     }
   }
 
+  function getCacheKey(keyTemplate: any): string {
```

Comment:
This function does not use anything of the internal state - we can turn it to an external function